### DISCUSSION TITLE
Databases and DB Admin tools

### DISCUSSION THREAD

**Discussion by Lockszmith-GH** [runtipi/runtipi-appstore#5350] [Ideas]
At first I thought it was by choice, not to include databases, as mostly they will be included in an app's stack.
But MongoDb exists, so that isn't it.
Also, PostgreSQL seems to be the preferred DB.

I would like to ask all of the major DBs be added:
- PostgresSQL
- MariaDB
- MySQL.

Also, admin WebUI's like:
- phpMyAdmin
- pgAdmin (additional ideas [here](https://pgdash.io/blog/postgres-gui-tools.html))

Seems like these would be great additions to the catalog.

**Comment by Lockszmith-GH** [comment]
After using runtipi for a bit, I now see that the 'appstore' metaphore currently allows only one instance per app - when it comes to backend infrastructure, that can be a drawback.

**Reply by AaronDewes** [reply]
Sounds not easy to make work from a technical and UX perspective (I've not looked at the Tipi code at all though, I'm just interested in the project).

I don't see an easy solution to make sure an app depending on another app connects to the "right" instance of that other app in case there are multiple instances.

**Reply by Lockszmith-GH** [reply]
if instance name is allowed to be changed, that instance name can be the hostname - very easy to solve.

It will mean that instance name, as well as exposed port # will need to be editable.
Not hard, and mostly will be left at default values.
